service/resource/podv2/delete.go

```diff
@@ -2,15 +2,116 @@ package podv2
 
 import (
 	"context"
+	"time"
 
+	"github.com/giantswarm/microerror"
 	"github.com/giantswarm/operatorkit/framework"
+	apiv1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	apismetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/kvm-operator/service/keyv2"
+)
+
+const (
+	PodDeletionGracePeriod = 0 * time.Second
```

Comment:
why we set grace deletion period to 0? that looks dangerous to me

Reply:
The actual grace period is 5 minutes. See https://github.com/giantswarm/kvm-operator/blob/f9692c8b03c4a0d3a448033d2e13a37655bd3024/service/resource/deploymentv2/worker_deployment.go#L19. There the right value is set to the worker deployment's pods. This here is when the pod got deleted and we want to delete it after draining. Then we set it to 0 to make Kubernetes really kill the processes so we can initialize the actual shutdown and garbage collection.

Reply:
But draining node does not equal machine being ready to be killed.

For example, another service could be running and it could possibly cause some data corruption in etcd.

I would rather have there still some period to allow a proper shutdown something like 1 minute at least.

Reply:
At this point here the services had 5 minutes to shutdown already. There is no need to define anything else than 0 at this point.

Reply:
OK